src/socket/meson.build

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023-2024 Rachel Mant <git@dragonmux.network>
+# SPDX-FileContributor: Written by Rachel Mant <git@dragonmux.network>
+
+rSONSocketLibs = []
+if host_machine.system() == 'windows'
+	rSONSocketLibs += cxx.find_library('Ws2_32')
+endif
+
+rSONSocketSrc = ['socket.cxx']
+
+rSONSocket = library(
+	'rSON.socket',
+	rSONSocketSrc,
+	cpp_args: ['-DrSON_SOCKET_EXPORT_API'],
+	include_directories: [rSONIncludeDir, rSONSocketIncludeDir],
+	link_with: [rSON],
+	dependencies: [rSONSocketLibs, substrate],
+	gnu_symbol_visibility: 'inlineshidden',
+	version: meson.project_version(),
+	install: true
+)
+
+if not meson.is_subproject()
+	pkgconfig.generate(
+		rSONSocket,
+		name: 'rSON Socket',
+		filebase: 'rSON_socket',
+		libraries: [rSON],
+		version: meson.project_version(),
+		description: 'Sockets handling for rSON, a simple but effective JSON parser for C++'
+	)
+else
+	rSONSocket_dep = declare_dependency(
```

Comment:
I think this should be unconditional, that way you can consume all dependencies through `dependencies` and not `link_with` (which only brings the lib, and not the includes).

Reply:
Ah, good catch - in other words, line 33 becomes the `endif` and the rest de-indented, popping the dependency down out of the conditional block, correct?

Reply:
We've implemented what we hope is an appropriate change for this - please give it a look over when you get a moment.